pegasus/sites.v3/hourofcode.com/views/course_carousel.haml

```diff
@@ -12,35 +11,26 @@
 
   %h3.centered{style: "font-size: 1.5rem; margin-block: 4rem 2rem"}!= heading
 
-  .learnpage{style: "position:relative"}
-    .carousel-wrapper
-      %div.carousel.course-carousel{id: id}
-        .slides
-          - items.each_slice(item_count) do |slice|
-            .slide.clear
-              - slice.each do |key, tutorial|
-                - url = tutorials.launch_url_for(key, request.host_with_port)
-                - url = "#{url}?company=#{company}" if company
-                .action-block-wrapper
-                  .action-block.action-block--one-col.white.flex-space-between
-                    .content-wrapper
-                      %p.overline-two= tutorial[:age]
-                      %h4{style: "font-size: 18px;"}
-                        = tutorial[:name]
-                      %img{src: tutorial[:image].sub("/images/", CDO.code_org_url("/images/fill-480x360/")).sub(".png", ".jpg"), alt: ""}
-                      %p
-                        %strong=hoc_s(:module_label_devices)
-                        = tutorial[:platformtext]
-                      %p.slide-longdescription= tutorial[:longdescription]
-                    .content-footer
-                      %a.link-button{href: url, target: "_blank"}
-                        =I18n.t(:go)
+  .carousel-wrapper
+    %swiper-container.three-col{id: id, navigation: "true", "navigation-next-el": ".swiper-nav-next.#{id}", "navigation-prev-el": ".swiper-nav-prev.#{id}", init: "false"}
+      - items.each do |key, tutorial|
+        - url = tutorials.launch_url_for(key, request.host_with_port)
+        - url = "#{url}?company=#{company}" if company
+        %swiper-slide
+          .action-block-wrapper
+            .action-block.action-block--one-col.white.flex-space-between
+              .content-wrapper
+                %p.overline-two= tutorial[:age]
+                %h4{style: "font-size: 18px;"}
+                  = tutorial[:name]
+                %img{src: tutorial[:image].sub("/images/", CDO.code_org_url("/images/fill-480x360/")).sub(".png", ".jpg"), alt: ""}
+                %p
+                  %strong=hoc_s(:module_label_devices)
+                  = tutorial[:platformtext]
+                %p.slide-longdescription= tutorial[:longdescription]
+              .content-footer
+                %a.link-button{href: url, target: "_blank"}
+                  =I18n.t(:go)
 
-        .button-wrapper
-          %button.slide-prev{href: "#", style: "border:none; background-color: var(--neutral_light);"}
-            %img{src: "//code.org/images/carousel/prev.svg", alt: "previous"}
-          %button.slide-next{href: '#', style: "border:none; background-color: var(--neutral_light);"}
-            %img{src: "//code.org/images/carousel/next.svg", alt: "next"}
-
-      .pagination
-        %div{id: "pagination-#{id}"}
+    %button.swiper-nav-prev{class: id}
+    %button.swiper-nav-next{class: id}
```

Comment:
where is `id` coming from? (for my own information)

Reply:
Tbh I don't know 😅 They show up magically in the DOM. I think this list might be coming from the tutorials somewhere, but I haven't looked into this code outside of making it work with Swiper.